weave/integrations/langchain/langchain_nv_ai_endpoints.py

```diff
@@ -0,0 +1,236 @@
+import importlib
+import time
+from collections.abc import Iterator
+from functools import wraps
+from typing import Any, Callable, Optional
+
+import_failed = False
+
+try:
+    from langchain_core.messages import AIMessageChunk, convert_to_openai_messages
+    from langchain_core.outputs import ChatGenerationChunk, ChatResult
+except ImportError:
+    import_failed = True
+
+import weave
+from weave.trace.op import Op, ProcessedInputs
+from weave.trace.op_extensions.accumulator import add_accumulator
+from weave.trace.patcher import MultiPatcher, SymbolPatcher
+
+
+# NVIDIA-specific accumulator for parsing the objects of streaming interactions
+def nvidia_accumulator(acc: Optional[Any], value: Any) -> Any:
+    if acc is None:
+        acc = ChatGenerationChunk(message=AIMessageChunk(content=""))
+    acc = acc + value
+
+    ## Need to do this since the __add__ impl for the streaming response is wrong
+    ## We will get the actual usage in the final chunk so this will be eventually consistent
+    acc.message.usage_metadata = value.message.usage_metadata
```

Comment:
What's wrong about the impl?

Reply:
it adds up all the tokens on a streaming response, rather than presenting the last value which is the actual amount of tokens

Reply:
Is it wrong just for this case, or for LC generally?

Reply:
seems to be lc generally